test/extended/syncer/syncer_utils.go

```diff
@@ -0,0 +1,142 @@
+	"fmt"
+	"strings"
+	"time"
+
+	o "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	exutil "github.com/kcp-dev/kcp-tests/test/extended/util"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+// SyncTarget struct definition
+type SyncTarget struct {
+	Name            string
+	SyncerImage     string
+	OutputFilePath  string
+	WorkSpaceServer string
+}
+
+// SyncTargetOption uses function option mode to change the default values of SyncTarget attributes
+type SyncTargetOption func(*SyncTarget)
+
+// SetSyncTargetName sets the SyncTarget's name
+func SetSyncTargetName(name string) SyncTargetOption {
+	return func(s *SyncTarget) {
+		s.Name = name
+	}
+}
+
+// SetSyncTargetOutputFilePath sets the SyncTarget's OutputFilePath
+func SetSyncTargetOutputFilePath(path string) SyncTargetOption {
+	return func(s *SyncTarget) {
+		s.OutputFilePath = path
+	}
+}
+
+// NewSyncTarget create a new customized SyncTarget
+func NewSyncTarget(opts ...SyncTargetOption) SyncTarget {
+	defaultSyncTarget := SyncTarget{
+		Name:            "mysyncer-" + exutil.GetRandomString(),
+		WorkSpaceServer: "",
+	}
+	for _, o := range opts {
+		o(&defaultSyncTarget)
+	}
+	return defaultSyncTarget
+}
+
+// Create SyncTarget
+func (s *SyncTarget) Create(k *exutil.CLI) {
+	if s.WorkSpaceServer == "" {
+		s.WorkSpaceServer = k.WorkSpace().ServerURL
+	}
+	s.CreateAsExpectedResult(k, true, "Wrote physical cluster manifest to")
+
+}
+
+// CreateAsExpectedResult creates SyncTarget CR and checks the created result is as expected
+func (s *SyncTarget) CreateAsExpectedResult(k *exutil.CLI, successFlag bool, containsMsg string) {
+	kcpServerVersion, getError := exutil.GetKcpServerVersion(k)
+	o.Expect(getError).ShouldNot(o.HaveOccurred())
+	o.Expect(kcpServerVersion).Should(o.MatchRegexp(`v\d+(.\d+){0,2}`))
+	msg, createError := k.WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().Run("kcp").Args("--server="+s.WorkSpaceServer, "workload", "sync", s.Name,
+		"--syncer-image=ghcr.io/kcp-dev/kcp/syncer:"+kcpServerVersion, "--output-file="+s.OutputFilePath).Output()
+	if successFlag {
+		o.Expect(createError).ShouldNot(o.HaveOccurred())
+		o.Expect(msg).Should(o.ContainSubstring(containsMsg))
+	} else {
+		o.Expect(createError).Should(o.HaveOccurred())
+		o.Expect(fmt.Sprint(msg)).Should(o.ContainSubstring(containsMsg))
+	}
+}
+
+// GetFieldByJSONPath gets specific field value of the SyncTarget by jsonpath
+func (s *SyncTarget) GetFieldByJSONPath(k *exutil.CLI, JSONPath string) (string, error) {
+	return k.WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().Run("get").Args("--server="+s.WorkSpaceServer, "syncTarget", s.Name, "-o", "jsonpath="+JSONPath).Output()
+}
+
+// CheckReady checks whether the SyncTarget is ready
+func (s *SyncTarget) CheckReady(k *exutil.CLI) (bool, error) {
+	readyFlag, err := s.GetFieldByJSONPath(k, `{.status.conditions[?(@.type=="Ready")].status}`)
+	if err != nil {
+		e2e.Logf(`Get SyncTarget/%s status faied: "%v"`, s.Name, err)
```

Comment:
```suggestion
		e2e.Logf(`Getting SyncTarget/%s status failed: "%v"`, s.Name, err)
```
And other places have similar typos to be updated.

Reply:
Addressed, thanks!